### ISSUE TITLE
Why the removalListener has not be invoked?

### ISSUE THREAD

**Issue by flykarry** [google/guava#3295]
![image](https://user-images.githubusercontent.com/14309127/46985918-4ebcfd80-d11f-11e8-910b-40b7d3c951dd.png)
![image](https://user-images.githubusercontent.com/14309127/46985958-76ac6100-d11f-11e8-8feb-570d725626ab.png)

**Comment by flykarry** [comment]
I knew to use cache.cleanup() to invoke the removallistener, but is there any function to let it run automatically？

**Comment by flykarry** [comment]
I knew to use cache.cleanup() to invoke the removallistener, but is there any function to let it run automatically？

**Comment by lowasser** [comment]
"If expireAfterWrite or expireAfterAccess is requested entries may be evicted on each cache modification, on occasional cache accesses, or on calls to Cache.cleanUp(). Expired entries may be counted by Cache.size(), but will never be visible to read or write operations."

Cleanups occur during regular usage of the cache, if you are actually making queries to it.  Otherwise you can create your own scheduled task on your own thread to do cleanup, but Guava does not attempt to do that for you, as that would require thread management which isn't available on all platforms (and isn't always advisable in the same way on all platforms).

**Comment by ben-manes** [comment]
This feature, of using an optional scheduling thread, is released in Caffeine 2.8.

```java
Caffeine.newBuilder()
    .scheduler(Scheduler.systemScheduler()) // Java 9+
    .expireAfterWrite(10, TimeUnit.SECONDS)
    .removalListener(...)
    .build();
```

If you are not on Java 9+ then you will have to supply your own thread, e.g. using a `Scheduler.forScheduledExecutorService(se)`.
